common/autoware_overlay_rviz_plugin/autoware_mission_details_overlay_rviz_plugin/src/mission_details_display.cpp

```diff
@@ -0,0 +1,245 @@
+#include <OgreMaterialManager.h>
+#include <OgreTechnique.h>
+#include <OgreTexture.h>
+#include <OgreTextureManager.h>
+
+#include <algorithm>
+#include <cmath>
+#include <iomanip>
+#include <memory>
+#include <mutex>
+#include <string>
+
+namespace autoware_mission_details_overlay_rviz_plugin
+{
+
+MissionDetailsDisplay::MissionDetailsDisplay()
+{
+  property_width_ = new rviz_common::properties::IntProperty(
+    "Width", 225, "Width of the overlay", this, SLOT(updateOverlaySize()));
+  property_height_ = new rviz_common::properties::IntProperty(
+    "Height", 100, "Height of the overlay", this, SLOT(updateOverlaySize()));
+  property_left_ = new rviz_common::properties::IntProperty(
+    "Left", 10, "Left position of the overlay", this, SLOT(updateOverlayPosition()));
+  property_top_ = new rviz_common::properties::IntProperty(
+    "Top", 10, "Top position of the overlay", this, SLOT(updateOverlayPosition()));
+
+  // Initialize the component displays
+  remaining_distance_time_display_ = std::make_unique<RemainingDistanceTimeDisplay>();
+}
+
+void MissionDetailsDisplay::onInitialize()
+{
+  std::lock_guard<std::mutex> lock(property_mutex_);
+
+  rviz_common::Display::onInitialize();
+  rviz_rendering::RenderSystem::get()->prepareOverlays(scene_manager_);
+  static int count = 0;
+  std::stringstream ss;
+  ss << "MissionDetailsDisplay" << count++;
+  overlay_.reset(new autoware_mission_details_overlay_rviz_plugin::OverlayObject(ss.str()));
+  overlay_->show();
+  updateOverlaySize();
+  updateOverlayPosition();
+
+  auto rviz_ros_node = context_->getRosNodeAbstraction();
+
+  remaining_distance_time_topic_property_ =
+    std::make_unique<rviz_common::properties::RosTopicProperty>(
+      "Remaining Distance and Time Topic", "/planning/mission_remaining_distance_time",
+      "autoware_internal_msgs/msg/MissionRemainingDistanceTime",
+      "Topic for Mission Remaining Distance and Time Data", this,
+      SLOT(topic_updated_remaining_distance_time()));
+  remaining_distance_time_topic_property_->initialize(rviz_ros_node);
+}
+
+void MissionDetailsDisplay::setupRosSubscriptions()
+{
+  // Don't create a node, just use the one from the parent class
+  auto rviz_node_ = context_->getRosNodeAbstraction().lock()->get_raw_node();
+
+  remaining_distance_time_sub_ =
+    rviz_node_->create_subscription<autoware_internal_msgs::msg::MissionRemainingDistanceTime>(
+      remaining_distance_time_topic_property_->getTopicStd(),
+      rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+      [this](const autoware_internal_msgs::msg::MissionRemainingDistanceTime::SharedPtr msg) {
+        updateRemainingDistanceTimeData(msg);
+      });
+}
+
+MissionDetailsDisplay::~MissionDetailsDisplay()
+{
+  std::lock_guard<std::mutex> lock(property_mutex_);
+  overlay_.reset();
+
+  remaining_distance_time_sub_.reset();
+
+  remaining_distance_time_display_.reset();
+
+  remaining_distance_time_topic_property_.reset();
+}
```

Comment:
It may be easier to read without new lines.

```suggestion
  remaining_distance_time_sub_.reset();
  remaining_distance_time_display_.reset();
  remaining_distance_time_topic_property_.reset();
}
```

Reply:
This was applied.